pennylane/ops/qubit.py

```diff
@@ -1997,7 +2033,15 @@ def decomposition(D, wires):
         return [QubitUnitary(np.diag(D), wires=wires)]
 
     def adjoint(self, do_queue=False):
-        return DiagonalQubitUnitary(self.parameters[0].conj(), wires=self.wires, do_queue=do_queue)
+        return DiagonalQubitUnitary(
+            qml.math.conj(self.parameters[0]), wires=self.wires, do_queue=do_queue
+        )
+
+    def _controlled(self, control):
```

Comment:
I really like the design with this function. Maybe we can make `adjoint` also private...

Reply:
+1 Will do that in another PR